ddtrace/contrib/psycopg/patch.py

```diff
@@ -65,13 +65,17 @@ def _trace_method(self, method, name, resource, extra_tags, dbm_operation, *args
         # treat psycopg2.sql.Composable resource objects as strings
         if isinstance(resource, Composable):
             resource = resource.as_string(self.__wrapped__)
-            # DBM context propagation is only supported for query args with the following type: string
-            dbm_operation = False
 
         return super(Psycopg2TracedCursor, self)._trace_method(
             method, name, resource, extra_tags, dbm_operation, *args, **kwargs
         )
 
+    def _dbm_sql_injector(self, dbm_comment, sql_statement):
+        if isinstance(sql_statement, Composable):
+            composable_dbm_comment = psycopg2.sql.SQL(dbm_comment)
+            return composable_dbm_comment + sql_statement
+        return super(Psycopg2TracedCursor, self)._dbm_sql_injector(dbm_comment, sql_statement)
```

Comment:
I'm not sure I understand this return, can you explain why we need to use super here?

Reply:
This function checks if the sql statement is a composable object. If it’s not then the query must be a string and the default dbm sql injector is used. The default sql injector is implemented in the parent class and we call it using super.  It simply adds strings. I can add a comment to make this more clear